dig.go

```diff
@@ -283,6 +268,26 @@ func newNode(provides reflect.Type, ctor interface{}, ctype reflect.Type) (node,
 	}, nil
 }
 
+// Retrives the dependencies for the parameter of a constructor.
+func getCtorParamDependencies(t reflect.Type) (deps []reflect.Type) {
+	if !t.Implements(_parameterObjectType) {
+		deps = append(deps, t)
+		return
+	}
```

Comment:
Are named returns providing much value here? To me, this is less readable than `return []reflect.Type{t}`.